### DISCUSSION TITLE
Can't upload regular jpg due to The field contains unknown filenames..

### DISCUSSION THREAD

**Discussion by janathaworth** [pocketbase/pocketbase#3609] [Q&A]
I am trying to upload a jpg almost exactly like the example shown here. https://pocketbase.io/docs/files-handling 

My data model includes: 

Collection named Post and inside the collection it has a column for a company logo. Which has all of the same properties as the one from the documentation. 
![company_logo_type](https://github.com/pocketbase/pocketbase/assets/20219434/3565f171-734d-4663-a4db-ffe1560141f4)

I am doing 

```
const createdRecord = await pb.collection('post').create(form.data); 
```

Where form.data.company_logo is of type File from (https://developer.mozilla.org/en-US/docs/Web/API/File) 

![fileexample](https://github.com/pocketbase/pocketbase/assets/20219434/5ac1b23a-6674-4669-ba2e-dc19d38de2d1)

Pocketbase logs says 
```

{
  "errorDetails": "company_logo: The field contains unknown filenames..",
  "errorMessage": "Failed to load the submitted data due to invalid formatting."
}
```

Any suggestions on what I might be doing wrong? Thanks!

**Comment by ganigeorgiev** [comment]
Most likely the `form.data` content is invalid (when uploading files the file field values must be either a `Blob` or `File` instance but in your case it seems to be submitted as a plain string).

Could you provide a code sample how you are constructing it?

**Comment by ronen** [comment]
In case anybody else finds this thread when searching for this error....

I've been getting the same error trying to add a local file to a `FormData` in nodejs.  In my case, for development & testing I want to seed a fresh database with some users by running a script .   The example at https://pocketbase.io/docs/files-handling only works in a browser.   What finally worked for me was:

```js
import { blob } from 'node:stream/consumers'
import fs from 'fs'
import PocketBase from 'pocketbase'

const pb = new PocketBase('http://127.0.0.1:8090');

const adminData = await pb.admins.authWithPassword('admin@example.com', 'admin-password');

const formData = new FormData()
formData.append('name', 'User Name')
formData.append('email', 'user@example.com')
formData.append('password', 'user-password')
formData.append('passwordConfirm', 'user-password')
formData.append('verified', true)
formData.append('avatar', await blob(fs.createReadStream('/path/to/file.png')))

const record = await pb.collection('users').create(formData)
```

**Reply by ganigeorgiev** [reply]
I don't think you need to create a stream of the file (_or at least I don't understand its purpose in this particular case_) and fetching the file content with `fs.readFileSync` should be sufficient (_or `fs.readFile` if you want to be async_).

See also https://github.com/pocketbase/pocketbase/discussions/1201.

_edit: Based on the Node docs, it seems that `createReadStream` will read the file on chunks but since you `await` the [`blob` function](https://nodejs.org/api/webstreams.html#streamconsumersblobstream) you'll end up with the entire content loaded in the memory anyway so I'm not sure if there will be any benefit._

**Reply by ronen** [reply]
The only purpose for the stream was that it worked :)    In chasing down how to validly get a file into `FormData` I had been following a lead that suggested using `fs.createReadStream` was the way to go, which led me to the way I did it.   My searching hadn't found #1201.   But yes you're right, this also works and is arguably cleaner behind the scenes:

```js
  formData.append('avatar', new Blob([fs.readFileSync('/path/to/file.png')]))
```
